authors/apps/profiles/views.py

```diff
@@ -15,32 +18,54 @@ class ProfileRetrieveUpdateView(generics.GenericAPIView):
     """
     Class for retriving  and updating user profile
     """
-    permission_classes = (permissions.IsAuthenticated,IsOwnerOrReadOnly)
+
+    permission_classes = (permissions.IsAuthenticated, IsOwnerOrReadOnly)
     renderer_classes = (ProfileRenderer,)
     serializer_class = ProfileSerializer
 
     def get(self, request, *args, **kwargs):
-        #Function to retrieve user profile
+        # Function to retrieve user profile
         serializer = self.serializer_class(
-            get_object_or_404(Profile, user=self.kwargs.get("username")))
+            get_object_or_404(Profile, user=self.kwargs.get("username"))
+        )
         return Response(serializer.data, status=status.HTTP_200_OK)
 
     def put(self, request, *args, **kwargs):
         # Function to update user profile
 
         try:
-            user = get_user_model().objects.get(username=self.kwargs.get("username"))
+            user = get_user_model().objects.get(
+                username=self.kwargs.get("username")
+            )
         except ObjectDoesNotExist:
             raise Http404()
 
         profile = Profile.objects.get(user=user.username)
 
         if not profile.user.pk == request.user.id:
-            return Response({"detail": "You don't have permissions to update this user"}, 
-            status=status.HTTP_403_FORBIDDEN)
+            return Response(
+                {"detail": "You don't have permissions to update this user"},
+                status=status.HTTP_403_FORBIDDEN,
+            )
         serializer = self.serializer_class(
-            profile, data=request.data["profile"], partial=True)
+            profile, data=request.data["profile"], partial=True
+        )
         if serializer.is_valid(raise_exception=True):
             serializer.save()
             return Response(serializer.data, status=status.HTTP_200_OK)
 
+
+class ListProfiles(APIView):
+    """
+    Class to list author profiles
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+    def get(self, request, *args, **kwargs):
+
+        articles = Article.objects.distinct("author").all()
```

Comment:
I also noticed that a user is not an author until they have published an article. So later on when we enable publishing articles, we shall filter only the published articles and filter against those.

Reply:
Great work @bisonlou and yeah @kalsmic that's a good observation I had planned to bring in the feature for publishing an article later on as for now, all articles have `publish_status=False`. We shall deal with this in technical debt.